crates/union_replacer/src/punning/transformation.rs

```diff
@@ -0,0 +1,319 @@
+                                                ident.unwrap(),
+                                                info.size,
+                                                info.size,
+                                            ));
+                                        }
+                                    }
+                                }
+                                if !found {
+                                    new_stmts.push(s.clone());
+                                }
+                            } else {
+                                new_stmts.push(s.clone());
+                            }
+                        } else {
+                            new_stmts.push(s.clone());
+                        }
+                    }
+                    _ => new_stmts.push(s.clone()), /* TODO: 다른 형태의 Init도 고려? LocalKind - Decl, InitElse */
+                },
+                // Writes
+                StmtKind::Expr(expr) | StmtKind::Semi(expr) => {
+                    if let Some((def_id, mir_locs)) = self.get_mir_func_locs_from_node(&expr.id) {
+                        let infos = self.transform_info.get(&def_id);
+                        if let Some(infos) = infos {
+                            let mut found = false;
+                            for info in infos {
+                                let write_locs = &info.write_locs;
+                                // TODO: Multiple Locations?
+                                assert_eq!(mir_locs.len(), 1);
+                                if let Some(replacable) = write_locs.get(&mir_locs[0]) {
+                                    found = true;
+                                    let ident = info.ident.as_ref().unwrap();
+                                    let rhs_expr = match &expr.kind {
+                                        rustc_ast::ExprKind::Assign(_, rhs_expr, _) => rhs_expr,
+                                        _ => unreachable!(),
+                                    };
+
+                                    let typecheck = self.tcx.typeck(def_id);
+                                    let hir_expr =
+                                        self.ast_to_hir.get_expr(rhs_expr.id, self.tcx).unwrap();
+                                    let expr_ty = typecheck.expr_ty_adjusted(hir_expr);
+
+                                    if *replacable {
+                                        // Replacable Write
+                                        new_stmts.push(utils::stmt!(
+                                            "{}_bytes = ({} as {}).to_be_bytes();",
+                                            ident,
+                                            pprust::expr_to_string(rhs_expr),
+                                            expr_ty.to_string()
+                                        ));
+                                    } else {
+                                        // Non-Replacable Write
```

Comment:
>  각 write에 대해, 다음과 같이 경우를 나누고 아래 변환 규칙을 적용
        (1) w를 replacable read들에서만 읽을 수 있는 경우
        (2) w를 non replacable read와 replacable read 모두에서 읽을 수 있는 경우
        (3) otherwise (w를 non replacable read에서만 읽을 수 있는 경우 또는 w가 어떠한 read에 대한 set에도 포함되지 않는 경우)
    Write 변환 규칙
        (1) byte array write으로 변환
        (2) byte array와 union에 대한 write이 공존
        (3) 원래대로 유지 (union write만)

지금은 3의 경우도 2로 처리해 버릴 수 있는 듯합니다.